src/lib.rs

```diff
@@ -1567,6 +1585,13 @@ pub enum TypeDefinitionProviderCapability {
     Options(StaticTextDocumentRegistrationOptions),
 }
 
+#[derive(Debug, Eq, PartialEq, Clone, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum HoverProviderCapability {
+    Simple(bool),
+    Options(HoverOptions),
+}
```

Comment:
You may want to add `impl From<bool/HoverOptions> for HoverProviderCapability`.

Reply:
@kjeremy Sure! Should we do the same for the other `.*Capability` enums used in `ServerCapabilities`?

Reply:
@kjeremy Done in 36c6012.